torch/distributed/elastic/examples/example_gpu_health_check.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python3
```

Comment:
Do you anticipate users manually invoking the health check server? I think if we integrate this in with torchrun that's all that's needed and then we can just document how to enable the feature rather than having a python example